docs/cstor.md

```diff
@@ -0,0 +1,384 @@
+| kubectl get cvr -n openebs         | Get the list of cStor volumes in the entire cluster          |
+| kubectl get volumesnapshot         | Get the list of volumesnapshots in the entire cluster        |
+
+
+
+## High availability of cStor
+
+cStor volumes when deployed in 3 replica mode provide high availability of the data as long as the replicas are in quorum. At least two replicas need to be healthy to call the volume is in quorum. In a 3 replica setup, if the second replica becomes unavailable because of the pool failure or unavailability, the volume is set to read-only by the target. When the volume replicas are back online, they are rebuilt one by one and the volume is set to read-write as soon as the quorum is achieved. 
+
+
+
+## Monitoring cStor pools and volumes
+
+The easiest way to monitor cStor pools and volumes is through MayaOnline. The volume metrics are scraped and uploaded to MayaOnline where users can browse historical volume performance. MayaOnline also provides the topology view where detailed live status of Volumes, snapshots, clones, pools and disks is obtained. Through the topology view , users get granular details of each of these Kubernetes resources in an intuitive graphical user interface. 
+
+
+
+Links to screenshots of some of the cStor resources in MayaOnline are shown below.
+
+[Pool topology view](/docs/next/mayaonline.html#cstor-pool-view)
+
+[Volume POD topology view](/docs/next/mayaonline.html#cstor-volume-pod-view)
+
+[Volume CR topology view](/docs/next/mayaonline.html#cstor-custom-resources-view)
+
+<br>
+
+## Thin provisioning in cStor
+
+cStor supports thin provisioning of volumes. By default, a volume is provisioned with whatever capacity that is mentioned in storage class. Capacity of the pool can be expanded on demand by adding more disks to the cStor pool. cStor architecture also supports the resizing of a provisoned volume on the fly, eventhough this feature is not fully functional yet and is part of the roadmap.
+
+<br>
+
+## Performance testing of cStor
+
+Performance testing includes setting up the pools, storage classes and iSCSI server tunables. Some best practices include 
+
+- Number of replicas - For statefulsets, when the application is doing the required replication, one replica at volume may be sufficient
+- Network latency - Latency between the pods and zones (if the replicas are placed across AZs) plays a major role in the performance results and it needs to be in the expected range
+
+### Seek support for help
+
+As cStor volume has a dedicated iSCSI stack, it can be tuned with number of worker threads and other advanced parameters depending on the type of workload.  
+
+- Connect the Kubernetes cluster to MayaOnline. It is free and provides seamless monitoring of OpenEBS volumes
+- Contact OpenEBS community on [Slack](https://slack.openebs.io) and seek further help
+
+
+
+## Known limitations
+
+**Kubernetes fails to decide if a node has come up or not :**
```

Comment:
After a node shutdown, I see Application stuck in container creating waiting for PV to be attached.